third_party/move/move-compiler-v2/tests/checking/inlining/lambda_cast.move

```diff
@@ -0,0 +1,9 @@
+module 0x12391283::M {
+    use std::vector;
+    fun main(gas_schedule_blob: vector<u8>) : u64 {
+        let gas_schedule_blob: vector<u8> = vector[
+            115, 115, 95, 112, 97, 99, 107, 101, 100, 32, 0, 0, 0, 0, 0, 0, 0,
+        ];
+        vector::fold<u64, u8>(gas_schedule_blob, (0 as u64), |sum, addend| sum + (addend as u64))
```

Comment:
What happens if we remove the explicit type parameter here?  `vector::fold(...)`

Also can we add a second test where this conversion is invalid to show that it generates an error?  For example, vector<vector<u8>>?
```
module 0x12391283::M {
    use std::vector;
    fun main(gas_schedule_blob: vector<u8>) : u64 {
        let gas_schedule_blob: vector<vector<u8>> = vector[
            vector[115], vector[115], vector[95], 
        ];
        vector::fold(gas_schedule_blob, (0 as u64), |sum, addend| sum + (addend as u64))
    }
}
```

Reply:
done